frontend/wasm/src/callable.rs

```diff
@@ -0,0 +1,75 @@
+use midenc_hir::{dialects::builtin::FunctionRef, interner::Symbol, Signature, SymbolPath};
+
+use crate::intrinsics::Intrinsic;
+
+/// Local core Wasm module functon or processed module import to be used for the translation of the
+/// Wasm `call` op.
+#[derive(Clone)]
+pub enum CallableFunction {
```

Comment:
This is so much nicer! Thank you! 
I remember trying the enum, but I did not like the result and ditched it half-way through. It appears that I was missing `Intrinsic`.